main/medium/src/reaper.rs

```diff
@@ -1696,6 +1696,116 @@ impl<UsageScope> Reaper<UsageScope> {
         PositionInQuarterNotes::new(qn)
     }
 
+    /// Returns the MIDI tick (PPQ) position corresponding to the start of the measure.
+    ///
+    /// # Safety
+    ///
+    /// REAPER can crash if you pass an invalid take.
+    pub unsafe fn midi_get_ppq_pos_start_of_measure(
+        &self,
+        take: MediaItemTake,
+        ppq: PositionInPpq,
+    ) -> PositionInPpq
+    where
+        UsageScope: AnyThread,
```

Comment:
Mmh. I'm not sure if for any of these functions `AnyThread` is okay. I think they are main-thread only. Would be great if you could ask Justin about this. If not, better default to main-thread-only (see https://github.com/helgoboss/reaper-rs/pull/67#issuecomment-1321198352). That includes:

- `UsageScope: MainThread`
- `self.require_main_thread()`

This functions still must remain unsafe because of the take pointer. But at least it would be *a bit* safer then :)